docs/built-in-models/encrypted_dataframe.md

```diff
@@ -0,0 +1,111 @@
+
+## Encrypt and Decrypt a DataFrame
+
+To encrypt a pandas `DataFrame`, use the `encrypt_from_pandas` function. First, a client engine, which manages keys, must be constructed using `load_client`.
+
+```python
+from concrete.ml.pandas import ClientEngine, load_encrypted_dataframe
+from io import StringIO
+import pandas
+
+data_left = """index,total_bill,tip,sex,smoker
+1,12.54,2.5,Male,No
+2,11.17,1.5,Female,No
+3,20.29,2.75,Female,No
+"""
+
+# Load your pandas DataFrame
+df = pandas.read_csv(StringIO(data_left))
+
+# Obtain client object
+client = ClientEngine(keys_path="my_keys")
+
+# Encrypt the DataFrame
+df_encrypted = client.encrypt_from_pandas(df)
+
+# Decrypt the DataFrame
+df_decrypted = client.decrypt_to_pandas(df_encrypted)
+```
+
+## Supported Data Types and Schema Definition
+
+Concrete ML's encrypted `DataFrame` operations support a specific set of data types:
+
+- **Integer**:  Integers are supported within a specific range determined by the encryption scheme's quantization parameters. Default range is 1 to 15. 0 being used for the `NaN`. Values outside this range will cause a `ValueError` to be raised during the pre-processing stage.
+- **Quantized Float**: Floating-point numbers are quantized to integers within the supported range. This is achieved by computing a scale and zero point for each column, which are used to map the floating-point numbers to the quantized integer space.
+- **String Enum**: String columns are mapped to integers starting from 1. This mapping is stored and later used for de-quantization. If the number of unique strings exceeds 15, a `ValueError` is raised.
+
+## Supported Operations on Encrypted Data-frames
+
+Encrypted DataFrames support a subset of operations that are available for pandas DataFrames. The following operations are currently supported:
+
+- `merge`: left or right join two data-frames
+
+<!--pytest-codeblocks:cont-->
+
+```python
+df_right = """index,day,time,size
+2,Thur,Lunch,2
+5,Sat,Dinner,3
+9,Sun,Dinner,2"""
+
+# Encrypt the DataFrame
+df_encrypted2 = client.encrypt_from_pandas(pandas.read_csv(StringIO(df_right)))
+
+df_encrypted_merged = df_encrypted.merge(df_encrypted2, how="left", on="index")
+```
+
+> **Important**: The merge operation on Encrypted DataFrames can be **securely** performed on a third-party server. This means that the server can execute the merge without ever having access to the unencrypted data. The server only requires the encrypted DataFrames.
+
+## Serialization of Encrypted Data-frames
+
+Encrypted `DataFrame` objects can be serialized to a file format for storage or transfer. When serialized, they contain the encrypted data and [evaluation keys](../getting-started/concepts.md#cryptography-concepts) necessary to perform computations.
+
+### Saving and loading Data-frames
+
+To save or load an encrypted `DataFrame` from a file, use the following commands:
+
+<!--pytest-codeblocks:cont-->
+
+```python
```

Comment:
I would import `from concrete.ml.pandas import load_encrypted_dataframe` here, else it's not clear where it comes from